app/src/main/java/fr/free/nrw/commons/customselector/data/MediaReader.kt

```diff
@@ -0,0 +1,71 @@
+package fr.free.nrw.commons.customselector.data
+
+import android.content.ContentUris
+import android.content.Context
+import android.provider.MediaStore
+import android.text.format.DateFormat
+import fr.free.nrw.commons.customselector.domain.model.Image
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import java.util.Calendar
+import java.util.Date
+import javax.inject.Inject
+
+class MediaReader @Inject constructor(private val context: Context) {
+    fun getImages() = flow {
+        val projection = arrayOf(
+            MediaStore.Images.Media._ID,
+            MediaStore.Images.Media.DISPLAY_NAME,
+            MediaStore.Images.Media.DATA,
+            MediaStore.Images.Media.BUCKET_ID,
+            MediaStore.Images.Media.BUCKET_DISPLAY_NAME,
+            MediaStore.Images.Media.DATE_ADDED,
+            MediaStore.Images.Media.MIME_TYPE
+        )
+        val cursor = context.contentResolver.query(
```

Comment:
Can you add a `// TODO:` comment - we also use Room in the app, and it supports returning a `Flow` [from a query](https://developer.android.com/training/data-storage/room/async-queries#flow-coroutines).  Eventually migrating this query to Room and returning the Flow directly would simplify this code, and for now, a `TODO` comment will serve as a nice breadcrumb trail toward a better overall architecture.

Reply:
Thanks for reviewing @psh. I doubt how Room can help fetch images from phone storage.